math/check_factorial.cpp

```diff
@@ -1,64 +1,72 @@
 /**
  * @file
- * @brief A simple program to check if the given number is a factorial of some
+ * @brief A simple program to check if the given number is a [factorial](https://en.wikipedia.org/wiki/Factorial) of some
  * number or not.
+ *
+ * @details A factorial number is the sum of k! where any value of k is a
+ * positive integer. https://www.mathsisfun.com/numbers/factorial.html
+ *
  * @author [Divyajyoti Ukirde](https://github.com/divyajyotiuk)
  */
-#include <cassert>
-#include <iostream>
+#include <cassert>   /// for assert
+#include <iostream>  /// for cout
 
+namespace math {
```

Comment:
Missing namespace Doxygen documentation.
```suggestion
/**
 * @namespace
 * @brief Mathematical algorithms
 */
namespace math {
```